src/basic/hashmap.c

```diff
@@ -1768,6 +1768,32 @@ int set_consume(Set *s, void *value) {
         return r;
 }
 
+int hashmap_put_strdup(Hashmap **h, const char *k, const char *v) {
+        int r;
+
+        r = hashmap_ensure_allocated(h, &string_hash_ops_free_free);
+        if (r < 0)
+                return r;
+
+        _cleanup_free_ char *kdup = NULL, *vdup = NULL;
```

Comment:
Ugh, I do not like this... Please move this to the beginning of the function.

Reply:
Well, by declaring this here, I'm telling the reader that those variables are not used before.
The compiler doesn't care. Why do you think it is better to move them to the top?

Reply:
I did not say it is better, but I prefer to move them to the beginning. I know we dropped that rule from our CODING_STYLE. So, it is a matter of taste. And this is your PR. I'll push "Resolve conversation" button.